core/modules/replica/Request.cc

```diff
@@ -0,0 +1,270 @@
+
+} /// namespace
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+std::string Request::state2string(State state) {
+    switch (state) {
+        case CREATED:     return "CREATED";
+        case IN_PROGRESS: return "IN_PROGRESS";
+        case FINISHED:    return "FINISHED";
+    }
+    throw std::logic_error(
+                    "incomplete implementation of method Request::state2string(State)");
+}
+
+std::string Request::state2string(ExtendedState state) {
+    switch (state) {
+        case NONE:                 return "NONE";
+        case SUCCESS:              return "SUCCESS";
+        case CLIENT_ERROR:         return "CLIENT_ERROR";
+        case SERVER_BAD:           return "SERVER_BAD";
+        case SERVER_ERROR:         return "SERVER_ERROR";
+        case SERVER_QUEUED:        return "SERVER_QUEUED";
+        case SERVER_IN_PROGRESS:   return "SERVER_IN_PROGRESS";
+        case SERVER_IS_CANCELLING: return "SERVER_IS_CANCELLING";
+        case SERVER_CANCELLED:     return "SERVER_CANCELLED";
+        case EXPIRED:              return "EXPIRED";
+        case CANCELLED:            return "CANCELLED";
+    }
+    throw std::logic_error(
+                    "incomplete implementation of method Request::state2string(ExtendedState)");
+}
+
+std::string Request::state2string(State state,
+                                  ExtendedState extendedState) {
+    return state2string(state) + "::" + state2string(extendedState);
+}
+
+std::string Request::state2string(State state,
+                                  ExtendedState extendedState,
+                                  replica::ExtendedCompletionStatus serverStatus) {
+    return state2string(state, extendedState) + "::" + replica::status2string(serverStatus);
+}
+
+Request::Request(ServiceProvider::pointer const& serviceProvider,
+                 boost::asio::io_service& io_service,
+                 std::string const& type,
+                 std::string const& worker,
+                 int  priority,
+                 bool keepTracking,
+                 bool allowDuplicate)
+    :   _serviceProvider(serviceProvider),
+        _type(type),
+        _id(Generators::uniqueId()),
+        _worker(worker),
+        _priority(priority),
+        _keepTracking(keepTracking),
+        _allowDuplicate(allowDuplicate),
+        _state(CREATED),
+        _extendedState(NONE),
+        _extendedServerStatus(ExtendedCompletionStatus::EXT_STATUS_NONE),
+        _performance(),
+        _bufferPtr(new ProtocolBuffer(serviceProvider->config()->requestBufferSizeBytes())),
+        _workerInfo(serviceProvider->config()->workerInfo(worker)),
+        _timerIvalSec(serviceProvider->config()->retryTimeoutSec()),
+        _timer(io_service),
+        _requestExpirationIvalSec(serviceProvider->config()->controllerRequestTimeoutSec()),
+        _requestExpirationTimer(io_service) {
+
+        _serviceProvider->assertWorkerIsValid(worker);

```

Comment:
this line should be indented 4, not 8

Reply:
Apologies from my editor (Komodo version 9) :-)